### PR TITLE
Implement zoom to mouse position on wheel event

### PR DESCRIPTION
**Pull request by JokerNN**
Demo

#72

### CODE REVIEW

**Comment by dylanvorster** [comment]
Oh noes! there is an issue when you translate an element, zoom and drag a link:)

**Comment by dylanvorster** [comment]
screen shot 2017-09-15 at 1 26 35 pm
ROFL XD

**Comment by JokerNN** [comment]
Oh well, looks like I hurried to much, :D need to rethink whole stuff

File: src/widgets/DiagramWidget.tsx [hunk 1]
```diff
@@ -287,11 +287,9 @@ export class DiagramWidget extends React.Component<DiagramProps, DiagramState> {
 			if (this.props.allowCanvasTranslation) {
 				diagramModel.setOffset(
 					this.state.action.initialOffsetX +
-						(event.clientX - this.state.action.mouseX) /
-							(diagramModel.getZoomLevel() / 100),
```
Comment (dylanvorster) [inline src/widgets/DiagramWidget.tsx]: Im almost certain that this is whats causing the issue, is it possible to move this down to the zoom logic as well?

Comment (dylanvorster) [reply src/widgets/DiagramWidget.tsx]: never mind xD

**Comment by dylanvorster** [comment]
Okay so the issue is that the order of matrix operations is now reversed. Instead of scaling and translating, we now translate and then scale. Working out the relative mouse position now needs to changed. Im trying to wrap my head around the formula changes.

**Comment by dylanvorster** [comment]
okay nearly there... its sort of working now

**Comment by JokerNN** [comment]
@dylanvorster so do I - I reverted `scale` and `transform` back, since it's not going to help all across appication and trying to calculate offsets with zoomLevels while scrolling. Main difficulty is that offset got scaled too

**Comment by dylanvorster** [comment]
awesome its fixed

**Comment by dylanvorster** [comment]
damnit, the selections are broken as well xD

**Comment by JokerNN** [comment]
😿  Sorry. That was really ambitious change

**Comment by dylanvorster** [comment]
I found another bug that isn't even related #FML

**Comment by dylanvorster** [comment]
RIGHT can you test the JokerNN-master branch

**Comment by JokerNN** [comment]
O my god, you're so smart 😲

**Comment by dylanvorster** [comment]
http://dylanv.blog/2017/09/15/storm-react-diagrams-3-1-0/

**Pull request merged**